src/mongocrypt-ctx-datakey.c

```diff
@@ -45,6 +47,135 @@ _next_kms_ctx (mongocrypt_ctx_t *ctx)
    return &dkctx->kms;
 }
 
+static bool
+_kms_kmip_start (mongocrypt_ctx_t *ctx)
+{
+   bool ret = false;
+   _mongocrypt_ctx_datakey_t *dkctx;
+   char *user_supplied_keyid;
+   _mongocrypt_endpoint_t *endpoint;
+   mongocrypt_status_t *status = ctx->status;
+   _mongocrypt_buffer_t secretdata = {0};
+
+   dkctx = (_mongocrypt_ctx_datakey_t *) ctx;
+
+   if (ctx->opts.kek.kms_provider != MONGOCRYPT_KMS_PROVIDER_KMIP) {
+      CLIENT_ERR ("KMS provider is not KMIP");
+      _mongocrypt_ctx_fail (ctx);
```

Comment:
Much as variable "ret" is set to false to indicate (wisely, I think) failure by default, should this also be indicated? Or perhaps always done in the goto target, if it's meant to set up failure any time the whole function fails. Rather than trying to set it each time; alternatively, maybe a fail: label that always correctly sets up failure is a good idea, or incoporating this behavior into label done: is a good idea. It looks to me like /you/ got it right in this function, hopefully the next person will be as attentive.